tls/extensions/s2n_client_key_share.c

```diff
@@ -56,6 +57,92 @@ const s2n_extension_type s2n_client_key_share_extension = {
     .if_missing = s2n_extension_noop_if_missing,
 };
 
+static int s2n_generate_preferred_key_shares(struct s2n_connection *conn, struct s2n_stuffer *out)
```

Comment:
We talked offline about this: I believe we can consolidate all these keyshare generating functions into one. They mostly differ in which keyshares are generated, which is controlled by the new bitfield.